### PR TITLE
LibJS: Infer file extensions when resolving module paths

### PR DESCRIPTION
**Pull request by linusg**

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/VM.cpp [hunk 1]
```diff
@@ -823,6 +823,29 @@ ThrowCompletionOr<void> VM::link_and_eval_module(Module& module)
     return {};
 }
 
+static String resolve_filepath(StringView filepath, StringView module_type)
+{
+    auto extensions = Vector<StringView> { "js"sv, "mjs"sv };
```
Comment (Hendiadyoin1) [inline Userland/Libraries/LibJS/Runtime/VM.cpp:828]:
```suggestion
    auto extensions = Vector<StringView,2> { "js"sv, "mjs"sv };
```
Lets not allocate here

File: Userland/Libraries/LibJS/Runtime/VM.cpp [hunk 2]
```diff
@@ -823,6 +823,29 @@ ThrowCompletionOr<void> VM::link_and_eval_module(Module& module)
     return {};
 }
 
+static String resolve_filepath(StringView filepath, StringView module_type)
+{
+    auto extensions = Vector<StringView, 2> { "js"sv, "mjs"sv };
+    if (module_type == "json"sv)
+        extensions = { "json"sv };
```
Comment (davidot) [inline Userland/Libraries/LibJS/Runtime/VM.cpp:830]: Maybe it is just me or that JSON modules are still new, but with json this behavior feels slightly more unexpected especially the `folder/index.json` case.

Not really a problem just something that I felt here.

**Review by davidot — APPROVED** [review]
LGTM, great to also have a test for it.

**Comment by linusg** [comment]
Whoever wrote this check messed up.

```
 Running Meta/check-newlines-at-eof.py... 
Traceback (most recent call last):
  File "/home/runner/work/serenity/serenity/Meta/check-newlines-at-eof.py", line 68, in <module>
    run()
[FAIL]: Meta/check-newlines-at-eof.py
Running Meta/check-png-sizes.sh... 
  File "/home/runner/work/serenity/serenity/Meta/check-newlines-at-eof.py", line 41, in run
    f.seek(f.tell() - 1, os.SEEK_SET)
ValueError: negative seek position -1
```

File: Userland/Libraries/LibJS/Runtime/VM.cpp (hunk 2 above)
Comment (linusg) [reply Userland/Libraries/LibJS/Runtime/VM.cpp:830]: I definitely want it for consistency, but I agree - it's less expected than for plain JS files.
FWIW, the codebase I work on at $work is also configured to do this (some webpack magic), so I'm quite used to seeing `import ... from "./Foo.fixtures";` which actually is `Foo.fixtures.json`.

File: Userland/Libraries/LibJS/Runtime/VM.cpp [hunk 3]
```diff
@@ -823,6 +823,29 @@ ThrowCompletionOr<void> VM::link_and_eval_module(Module& module)
     return {};
 }
 
+static String resolve_filepath(StringView filepath, StringView module_type)
```
Comment (IdanHo) [inline Userland/Libraries/LibJS/Runtime/VM.cpp:826]: maybe resolve_module_filepath? This is a pretty generic name

File: Userland/Libraries/LibJS/Runtime/VM.cpp (hunk 2 above)
Comment (davidot) [reply Userland/Libraries/LibJS/Runtime/VM.cpp:830]: Yeah it seems fine, just JSON modules being new to me

File: Userland/Libraries/LibJS/Runtime/VM.cpp (hunk 3 above)
Comment (alimpfard) [reply Userland/Libraries/LibJS/Runtime/VM.cpp:826]: \*ahem\*, `filepath` isn't a word, can we have `file_path` pls?

Comment (linusg) [reply Userland/Libraries/LibJS/Runtime/VM.cpp]: ACK

**Pull request merged**